test: test_backtest_to_live_transition 预存在失败 (engine_service.add 返回非成功)

## 现象
`tests/unit/data/services/test_engine_service.py:476` 预存在失败：
```python
def test_backtest_to_live_transition(self, engine_service):
    create = engine_service.add(name="integration_transition", is_live=False)
    assert create.is_success()  # ← AssertionError: False
```

## 复现
```bash
.venv/bin/python -m pytest "tests/unit/data/services/test_engine_service.py::TestEngineServiceIntegration::test_backtest_to_live_transition" -o addopts=""
# 1 failed
```

## 影响范围
- 预存在于 master（commit 24a66277），stash 改动后单跑仍红，**非 PR#6232 回归**。
- TestEngineServiceIntegration 属集成测试（用真实 engine_service fixture），失败点在第一步 engine_service.add() 即返回非成功，疑为 fixture/DB 接线或 add() 逻辑问题，待 triage 区分环境 vs 真实 bug。

## 发现时机
PR#6232 (DI 容器 + search_logs 修复) 回归时，stash 对比确认与本 PR 无关。

--- a/tests/unit/data/services/test_engine_service.py
+++ b/tests/unit/data/services/test_engine_service.py
@@ -466,14 +466,14 @@ class TestEngineServiceIntegration:
         verify = engine_service.get(engine_id=uuid)
         assert len(verify.data) == 0
 
-    def test_backtest_to_live_transition(self, engine_service):
+    def test_backtest_to_live_transition(self, engine_service, unique_name):
         """测试回测到实盘的转换流程"""
         # 1. 创建回测引擎
         create = engine_service.add(
-            name="integration_transition",
+            name=f"integration_transition_{unique_name}",
             is_live=False
         )
-        assert create.is_success()
+        assert create.is_success(), f"Create failed: {create.error}"
         uuid = create.data["engine_info"]["uuid"]
 
         # 2. 升级为实盘

